cmd/generate_repo_config/generate_repo_config_test.go

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2016 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
```

Comment:
I'd rather see an integration test of the behavior, rather than a test of `generateRepoConfig` specifically. Perhaps this could be added as test case in `//internal:go_repository_test`: run a build command on `@bazel_gazelle_go_repository_config//:all`, then read and verify the configuration file in `$(bazel info output_base)/external/bazel_gazelle_go_repository_config/`.